### DISCUSSION TITLE
障害報告スレ

### DISCUSSION THREAD

**Discussion by snext1220** [snext1220/stext#303] [Q&A]
ソーサリアン Text創作／プレイ時の障害報告のためのスレッドです。
「バグかな？」と思った時には、こちらのスレから状況お知らせいただければ幸いです。

**以下の障害時マニュアルもご確認ください。**
**https://sorcerian.hateblo.jp/entries/2018/06/18**

なお、投稿に際しては、以下のテンプレートに沿って情報提供戴くと、問題解決がスムーズになります。
ご協力戴けますと幸いです。

---

+ 使用しているOS

+ ブラウザーの種類／バージョン

+ 生じた問題の詳細（そこまでの手順も）

+ ログ／ストレージのキャプチャ（詳細は上のURLを確認ください）

＜※この部分に画像をドラッグ＆ドロップします＞

---

**Comment by RYU-DS** [comment]
パラメータ設定したHPを戦闘で増減させた結果が条件式に反映されないようです。
具体的にはモスのHPをp2861に設定、シーン2867の戦闘で減らし（画像1-2）、次のシーン2869でそれが0じゃない場合「ドンハー～（!p2861:@0）」を表示）、0の場合「ヒィーハァー（p2861:@0）」を表示するようにしてるのですが（画像3-4）、0であってもドンハーのみが表示されます（画像5）。
リロードするとその通りに表示されるので式は間違っていないようですが…（画像6）
2867と2869の間にもう一つシーンを挟んでみてもダメでした（画像7）。
HP0のままシーンをループして戻ってきても同じ、さらにHP0でセーブされてる状態で終了して再開しても同じ。
2867でHPクリックで0＞リロードではクリック前の数値に戻るので、上のParamsボックスのp2861を0にしてリロードしてHP0の状態で2869へ移動するとヒィーハァーが表示されるので、とにかくデバッグ実行でのリロードをしない限り反映されないようです。
※追記：間に挟んだ2870でシーン要素で0にしてもダメでした…
画像1
![スクリーンショット 2023-01-28 042329](https://user-images.githubusercontent.com/43634197/215194786-415d5a20-0a2e-4065-b745-e62505c8b887.png)
画像2
![スクリーンショット 2023-01-28 042436](https://user-images.githubusercontent.com/43634197/215194497-d6e4abb3-f45a-4c85-9545-eeeb12821880.png)
画像3
![スクリーンショット 2023-01-28 054632](https://user-images.githubusercontent.com/43634197/215194876-4d566657-a6d9-4ad3-ae3b-87c17abb0840.png)
画像4
![スクリーンショット 2023-01-28 054657](https://user-images.githubusercontent.com/43634197/215194939-59d70475-54b5-4825-bbb2-ec001a9eb836.png)
画像5
![スクリーンショット 2023-01-28 043546](https://user-images.githubusercontent.com/43634197/215195084-21ada235-a93f-4d5c-b113-a868a349a687.png)
画像6
![スクリーンショット 2023-01-28 043616](https://user-images.githubusercontent.com/43634197/215195190-f4bb0da0-6c0c-4f95-86e4-f2c7d24a3520.png)
画像7
![スクリーンショット 2023-01-28 043710](https://user-images.githubusercontent.com/43634197/215195306-5c387375-ed04-4968-96e6-282c2e94208b.png)

**Comment by RYU-DS** [comment]
わかりにくいのでTest用シナリオ作りました。※画像１
![スクリーンショット 2023-01-30 001208](https://user-images.githubusercontent.com/43634197/215336527-db2af255-9a65-4c77-a211-5748edec8d1b.png)

①HP10のまま※画像２→「HP0以外」「HP1以上」が表示※画像３
![スクリーンショット 2023-01-30 001621](https://user-images.githubusercontent.com/43634197/215336576-4603d89b-7c87-417e-818d-0f1defdb8e44.png)
![スクリーンショット 2023-01-30 001715](https://user-images.githubusercontent.com/43634197/215336615-812c4398-7493-462b-b77f-084dddc961b6.png)

②HP0にしても※画像４→同じく「HP0以外」「HP1以上」が表示※画像４
![スクリーンショット 2023-01-30 001811](https://user-images.githubusercontent.com/43634197/215336640-5ed85d85-59f5-4238-a2b0-2f207c0f32ee.png)
![スクリーンショット 2023-01-30 001834](https://user-images.githubusercontent.com/43634197/215336655-6174ac60-e4c3-431e-9b91-9f16ba50a5d0.png)

③②の状態で右上でRelodすると「HP0」「HP1未満」が表示※画像５
![スクリーンショット 2023-01-30 001903](https://user-images.githubusercontent.com/43634197/215336676-429b65d6-915b-4457-835a-ae10f4d8e04f.png)

※キーに書いてある式は元の半角だとキーが表示されないので全角にしてあります。

**Reply by toki-sor1** [reply]
@RYU-DS さん、ご質問を拝見いたしました！
状況の詳しいご説明やtestシナリオのご作成等、ありがとうございます＆お疲れ様でした！
ご不便をお掛けしてしまい、申し訳ありません……！

私もRYUさんのtestシナリオを参照させていただき、testシナリオを作成して状況再現とテストを行いました。
内部的な処理といった詳しいことは、私もちょっと分からなくて自信がないのですが（汗）、testシナリオを例に解決方法を考えてみました。

1. 「内部パラ」タブのidに「p01」、初期値に「0」を入力
2. シーン1「敵戦闘」の内部パラに「p01:10」と入力
3. あとの設定は、そのままでOK

![image](https://user-images.githubusercontent.com/42646342/215493176-26db76d1-a029-4eb1-80bd-c81493b63b08.png)

![image](https://user-images.githubusercontent.com/42646342/215494271-ce93c1fc-e2a6-44c7-bb68-6238bf4cffbc.png)

![image](https://user-images.githubusercontent.com/42646342/215493699-2731dbb9-af50-4fa2-9d1d-c1d2e35d7acb.png)

おそらく**p01の元々の値がシーン1で設定されていなかったこと** と **敵のHPとなるp01の初期値を、内部パラのタブから10に設定していたこと**で、分岐先が正常に表示されなかったのかもしれません。

もしよろしければ、こちらのtestシナリオをPlaygroundにインポートしてご参照くださいませ～。
**（追記：すみません、古いファイルの方をアップしてしまいました；；　修正版をアップ済みです……！）**
[p_test.zip](https://github.com/snext1220/stext/files/10536209/p_test.zip)

**Comment by RYU-DS** [comment]
解決しました！ありがとうございます！

しかし初期値を見てるんであれば今度は逆にHP0の時のキーしか表示されないはずだと思ってたんですが…

なお元々のそのスレの方では画像のように設定して特にシーンでは数値を指定していませんでしたが
![スクリーンショット 2023-01-30 235322](https://user-images.githubusercontent.com/43634197/215511114-c32eab4b-5b12-467a-a27a-5ac12a68395a.png)
これを初期値0にして戦闘前のシーンで999に指定したらOKになった、という事は
内部パラは初期値を0に設定して必要なシーンで数値を入れるようにしないと正常に動作しない、という認識でいいのでしょうか？

**Reply by toki-sor1** [reply]
@RYU-DS さん、お返事ありがとうございます！
問題も解決したとお聞きして安堵しました！　お疲れ様です～！！

> しかし初期値を見てるんであれば今度は逆にHP0の時のキーしか表示されないはずだと思ってたんですが…

う～ん、必要なシーンの内部パラで設定した数値がダメージを受けて0になったことで、**初期値0に戻ってHP0の時のキーが反映されるという感じなのでしょうかね……？**

（**試しに初期値100、必要なシーンの内部パラ0**にしてみたところ、**分岐が正常に表示されませんでした**）
※間違えて100で試していたので10に修正してから試しましたが、こちらも同じ結果でした

> これを初期値0にして戦闘前のシーンで999に指定したらOKになった、という事は
内部パラは初期値を0に設定して必要なシーンで数値を入れるようにしないと正常に動作しない、という認識でいいのでしょうか？

はい～、 **おそらくはRYUさんのご認識のとおりだと思います～。**
私も皆さんのシナリオでParamがどのような使われ方をしているのかを拝見し、そこから色々と試して「あ！できた！！」となったので、内部ではどういう処理が行われているのか分からないのですが、testシナリオを例にまとめると、

- 敵HPでParamを扱う場合、「内部パラ（p01,初期値0）」タブ＋「敵（p01）」タブ＋「必要なシーン（シーン1）」で**データを紐づける必要がある**
- 「内部パラ」タブの初期値は **「0」** にして、必要なシーンに **「id＋敵HPの数値」** を入れないと、分岐が正常に表示されない
- 必要なシーンに「p01:10」と設定することで、初めて**シナリオ内で正式（？）に敵HP10（p01:10）のデータが反映される？**

という感じなのかもですね。
正常に動作しなかった原因の1つと思われるのですが、RYUさんにご提供いただいたシーン1の画像と私のシーン1の画像を見比べると、**RYUさんの画像のシーン1には「p01:10」が表示されていない**という違いもありました。
**シーン1にp01のデータが反映されていないままp01の処理を行おうとして不具合が出ていたのかもしれませんね。**

ぐぬぬ、原理が分からず何だかはっきりとしないお返事になってしまって恐縮です；；
前回から更にパワーアップした「そのスレ」を楽しみにしております～！

**Comment by RYU-DS** [comment]
＞ぐぬぬ、原理が分からず何だかはっきりとしないお返事になってしまって恐縮です；；
確かにちょっとスッキリしない所はありますが、対処療法でもとにかく使えるようになったので助かりましたー

しかしこういう事ができるとさらに欲が出てきますね…
Paramが戦闘式に組み込めるようになれば、特殊な戦闘でも大体の事は自動化できて手動要素がずいぶん減るんですが、それはそれでゲームブックっぽさも薄れますから痛し痒しといった所ですかね

**Comment by RYU-DS** [comment]
敵HPをパラメータ設定、戦利品でパラメータ増減、というのをやってみたのですがどうにも反映されません。
以前の失敗からHPは初期値0でそのシーンで満タンにしてます。
dropはparam/p10911:10/HP回復で、10、-10、@10どれも全く反応が無いので未実装なのか？と思ったりするのですが…
どなたか戦利品でのパラメータ増減実際できたという方いらっしゃいますでしょうか？

![スクリーンショット 2024-06-09 223727](https://github.com/snext1220/stext/assets/43634197/8a8b5add-6690-4bf7-9e2c-0c18d99ed6ad)

![スクリーンショット 2024-06-09 223809](https://github.com/snext1220/stext/assets/43634197/63619afe-0ae8-49b8-9370-6a1e8f51eb61)

![スクリーンショット 2024-06-09 223946](https://github.com/snext1220/stext/assets/43634197/2e072b49-60fc-414c-8ec5-0e68b552936a)

**Reply by toki-sor1** [reply]
@RYU-DS さん、シナリオ制作お疲れ様です＆Playgroundのご利用ありがとうございます！
ご質問拝見いたしました！（スクショを拝見して「あ！　F○3だ！」と懐かしくてはしゃいでいましたｗ）
私はシステム面での内部処理は専門外のため、ご利用方法や意図している内容と異なった回答でしたら申し訳ありません……！

### 戦利品でパラメータ増減
**パラメータ（この場合は敵HP）を次のSceneに持ち越しさせる前提**または**再読み込み→続きからプレイ再開させる前提**であれば**戦利品で増減させることは可能**でした。

### 内部パラの設定
そのままの設定（最小値0、最大値555、初期値0）で大丈夫だと思います。

### 敵の設定
下記の設定のみ変更していただけますでしょうか？
- 【ドロップ】param/p10911:10/HP回復
- 【敵HP】p10911

### シーンの設定
Scene 6003（該当の敵が登場するScene）で下記の設定を試していただけますでしょうか？
- 【内部パラ】p10911:555
- HP持ち越し前提でしたら、次のSceneにも必ず敵idの入力をお願いします
  - 再読み込み→続きからの手順を挟む場合は、次のSceneでの敵id入力は不要です
***
色々と試してみましたが、paramを利用した戦利品でのパラメータ増減は、（私が見付けられた範囲では）下記の方法で一応は可能なようでした。
- **戦利品ボタンを押したSceneから次のSceneへと増減させたパラメータを持ち越す場合**は、次のSceneにも敵idを設定する
- **戦利品ボタンを押したSceneでパラメータを増減させる場合**は、再読み込み→続きからの手順が必要になる

回答になっているのか分かりませんが、少しでも参考になりましたら幸いです～。

**Comment by RYU-DS** [comment]
あー…次シーンでの反映でしたか…
敵HPはクリックで即減るので現シーン反映だと思い込んでました、すみません。
次シーン移動であれば、戦利品を使うまでも無くパラ増シーンへ移動しての敵HP回復ができますが、あまり現実的でないので…
やはりボス戦以外での敵HP回復は難しそうですねーといっても実は戦闘が助長になるためあまり導入したくないので問題無かったりします（不具合の可能性を考えて報告しましたが、短慮でした）。
お手数おかけました。(´・ω・)ｽﾏｿ

＞「あ！　F○3だ！」
最近こういうの考えだしたら止まらなくなってこんなんばっかり考えてますｗ
最近リメイクでWIZが話題になると裸忍者出したくなったりｗ

**Comment by RYU-DS** [comment]
以前からおかしいと思ってたんですが、ハッキリわかったので報告します。

シーン文章の最後に入れた改行（通常1行以上は1行に、スペース2つ入で2行以上も可）がEditerからFlowに移動すると消えるようです。

以前から入れたハズなのに気付くと無くなってて「おかしいな？」と思ってんですが…
何行入れようが、すべてのシーンの全部が消えます。

当方では最後の文章とキー表示文章をつなげるようにしている関係でこの隙間はコントロールしたいのですが…
対策としては「最後に罫線などを入れてただの隙間ではなくす」というのがあります。
自分としてはあまりしっくりきませんでしたが…

あとついでに罫線ですがFlow説明にある====では出ず、MarkDown説明の方にあるのもどれで出るかが前後の条件などで出たり出なかったりメチャクチャ複雑なようです。一応***が安定して出るようです。

**Reply by toki-sor1** [reply]
@RYU-DS さん

この度は、ご報告ありがとうございます～！
改行について＆罫線について、こちらでも検証してみました！

> シーン文章の最後に入れた改行（通常1行以上は1行に、スペース2つ入で2行以上も可）がEditerからFlowに移動すると消えるようです。

こちら、私の方でもEditerからFlowに移動するとシーン文章の改行が消えていることを確認いたしました。
確か私もかなり昔に同じことで困ったのですが、以下の方法で解決できそうです。

![image](https://github.com/user-attachments/assets/c554cae3-75ff-41eb-8d10-156203562acb)
```
上の図と同じように<br>を横並び 又は 縦並びに同じように<br>を増やしていくと、改行が2行、3行と増えていきます。 
```
この方法だと、**EditerからFlowに移動してもシーン文章の最後の改行が消えない**ので安心です。
参照URL：[クロノドライブのHTML辞典](https://html-coding.co.jp/annex/dictionary/html/br/)
***
> あとついでに罫線ですがFlow説明にある====では出ず、MarkDown説明の方にあるのもどれで出るかが前後の条件などで出たり出なかったりメチャクチャ複雑なようです。一応***が安定して出るようです。

こちらも====では罫線が表示されないことを確認いたしました。
前後の条件については、条件が分からず検証できなかったのですが、RYUさんが仰る通り***で罫線が表示されますね。
```
「*」や「 _（アンダーバー）」が1つだけでも罫線が表示されたのが不思議なのですが、
「***」や「___（アンダーバー3つ）」辺りが良いのかもしれません。
```
※下記サイトの表記例の中でも、「---」「_ _ _（半角スペース有りアンダーバー3つ）」は罫線が表示されませんでした
参照URL：[NotePM](https://notepm.jp/help/how-to-markdown#%E6%B0%B4%E5%B9%B3%E7%B7%9A%EF%BC%88%E7%BD%AB%E7%B7%9A%EF%BC%89)

これで問題が解決できていたら良いのですが、どうでしょうか～。
（一部文字が小さくなって読み辛いところがあると思うのですが、投稿後の文章がHTMLやMarkdown記法により正常に表示されなくなってしまうため無効化しています；

**Comment by RYU-DS** [comment]
おぉーbreakコマンド？みたいなものがあるんですねー
PGEditerというかSGMLでもともと使うコマンドなんですかね？

実は投稿直後に力技で自己解決したんですがその方法は…
%white%　%/%
を最後に入れる事です。
クリックで反転させるとスペースがあるのがわかってしまいますがｗ
どっちにしろ数百～千以上のシーンに今から入れる気力は無いので頑張って溜めないと…

罫線はFlow説明の中のMarkdown記法のページにある
![スクリーンショット 2024-12-05 065846](https://github.com/user-attachments/assets/a7362a55-d902-4fcd-955f-e5ac2622cf30)
これをコピペでシーン文章に入れて、数を増減したり前後に別の文字を入れたりすると、もう面白いくらいメチャクチャになりますｗ

**Reply by toki-sor1** [reply]
@RYU-DS さん

お返事が遅くなり申し訳ありません……！

> PGEditerというかSGMLでもともと使うコマンドなんですかね？

SGMLは元々XMLベースのマークアップ言語のようで、STextでも改行時にはXMLファイルで有効な<ｂｒ>または<ｂｒ />が使用できるのではないかと推測しております～。

> %white%　%/%

私も以前、%white%・%/%という表記で空白を作っていたことがあるので懐かしくなりましたｗ
確かに数百～千以上のシーンに空白を入れ直すのは辛過ぎますね……お、お疲れ様です……！

> もう面白いくらいメチャクチャになりますｗ

試してみたら、条件によっては罫線が表示されたり太文字になったり等、文章がメチャクチャになっちゃいましたｗ
どうしてそうなってしまうのかは不明ですが、やっぱりこういう表記等は複雑かつ繊細そうですね～。

**Comment by RYU-DS** [comment]
画像のシーン埋め込みが上手くいかず四苦八苦してたのですが、わかってしまえばデスヨネーな件だったので警鐘がてら報告します。

![スクリーンショット 2024-12-21 012903](https://github.com/user-attachments/assets/e689d35e-4da8-4609-b2ef-457709bd9f99)

「map01」という画像を埋め込もうとして「![★MAP：街](map01)」をシーンに入れて画像もちゃんとcapturフォルダに入れてたのですが、どうにも表示されず…Markdown説明サイトを見てやっと解決しました。

「![★MAP：街](map01.png)」とする必要があったんですね…Markdown説明サイトを見れば何のファイルを見に行くか指定する必要があるとわかるのですが、上記画像の通りPGFlowの説明では「(ファイル名)」となっているので「(map01)」であってるハズなんですよ…

まさかPGFlowの説明が間違ってるとは思わないので（このコマンドで自動的にcapturフォルダ内のpngファイルのみからmap01を見に行くものだと思ってました）、他にミスがあるんだろうといろいろ試しました。

既にこの機能を使ってるほぼ全員の方々には当たり前の事ですが、やった事無い人間がやるとこの通りなので「(ファイル名+拡張子)」にするなど修正した方がいいと思います。
これから新規の人がこの機能を使う機会があるかどうかはわかりませんが

**Reply by toki-sor1** [reply]
@RYU-DS さん

改めて、ご報告ありがとうございました！
また、ご不便をお掛けしてしまい申し訳ありません……！

画像ファイルの埋め込みに関して、私も以前にファイル名の表記に誤りがあり、画像が表示できなくて困っていたことがあったような覚えがあります。
確か、私も原因が同じだったような気がするのですが……すみません、ちょっと記憶が曖昧です；

ただ、ご指摘のとおり「（ファイル名）」という説明では「（map01）」のように拡張子を入れなくても良い、という意味としても受け取れてしまうので **「(ファイル名+拡張子)」の方がより正確で分かりやすいですね！**

私ではマニュアルの編集ができないもので、すぐに修正できず大変恐縮なのですが、心に留めておきますね～！

**Comment by RYU-DS** [comment]
実は今日仕事先のPCで作業をしてる時、ファイルの名前を変更しようとしたら拡張子込みで変更するよう表示されて「あれ？」となって、OSのVerで違う？とか思って帰ってきて見てみたら…

![スクリーンショット 2024-12-24 010938](https://github.com/user-attachments/assets/8acd0744-2722-4185-a82f-5dfadf4ba828)

![スクリーンショット 2024-12-24 011007](https://github.com/user-attachments/assets/8b6acfb8-de73-4eab-b620-2d697f473d47)

普通にこういう表示選択があるくらい、ファイル名には拡張子込も普通なんですね…
言われてみれば昔少しですがPC98触ってた時はファイル名自体がDATA.EXEとかだった気がします。
拡張子無しは近年のwindowsとか用の表示なんですね、すっかり失念してました。
OLDPCゲーマーとして恥ずかしい…orz

**Reply by toki-sor1** [reply]
どんまいです！
はいー、以前のWindowsだとファイル名が拡張仕込みで表示されていましたが、近年のWindowsでは、拡張子無し状態が標準になっていますよね〜。

私は拡張子が表示されていないと何の形式のファイルなのかすぐに判断できなくて困ることが多かったので、常に拡張子込みに設定し直してました；

調べてみると、Windows10辺りからファイル名の拡張子無しが標準となったようですね。
その理由も「拡張子を誤って消してしまうのを防ぐため」のようで、確かにファイル名の変更時に拡張子まで消してしまうと、ファイルが正常に開けなくなってしまうので、標準で非表示になったのも納得だなぁ、と！
勉強になりました！！
